__tests__/api/chat/channels.test.js

```diff
@@ -20,31 +46,88 @@ describe('GET /chat/channels/signed', () => {
       .set({Authorization: `Bearer valid-token`})
       .expect(200);
 
-    expect(response.body).toEqual(
-      expect.objectContaining({
-        code: 200,
-        data: expect.arrayContaining([
-          expect.objectContaining({
-            members: expect.arrayContaining([expect.any(Object)]),
-            messages: expect.any(String)
-          })
-        ]),
-        status: 'Success retrieve channels'
-      })
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 0)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      2,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 30)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      3,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 60)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      4,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 90)
+    );
+  });
+
+  test('should return 200 OK with list of channels with offset only', async () => {
+    const user = await createUser();
+
+    const response = await request(app)
+      .get('/api/v1/chat/channels/signed?offset=10')
+      .set({Authorization: `Bearer valid-token`})
+      .expect(200);
+
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 10)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      2,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 40)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      3,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 70)
     );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      4,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 100)
+    );
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, avoid making assumptions about the internal implementation of the `queryChannels` method.



<!-- This is an auto-generated comment by OSS CodeRabbit -->